### PR TITLE
arith proof rules shuffle & add ARITH_SUM_UB

### PR DESCRIPTION
**Pull request by alex-ozdemir**
Preparation for making ARITH_SCALE_SUM_UB a macro.

* Adds a proof rule for summing upper bounds: ARITH_SUM_UB.
* Moves ARITH_MULT_* rules from the non-linear extension to the main
  arithmetic checker, since they will be needed for all of arith now.
* Aligns the ARITH_SCALE_SUM_UB documentation with its checker.

### CODE REVIEW

File: src/expr/proof_rule.h [hunk 1]
```diff
@@ -1043,15 +1043,23 @@ enum class PfRule : uint32_t
...
   //    its ki is negative). >< is always one of <, <=
   //    NB: this implies that lower bounds must have negative ki,
   //                      and upper bounds must have positive ki.
-  //    t1 is the sum of the polynomials.
-  //    t2 is the sum of the constants.
+  //    t1 is the sum of the scaled polynomials (sum_i k_i poly_i)
+  //    t2 is the sum of the scaled constants. (sum_i k_i const_i)
   ARITH_SCALE_SUM_UPPER_BOUNDS,
 
+  // Children: (P1, ... , Pn)
```
Comment (ajreynol) [inline src/expr/proof_rule.h:1055]: For style, this should have a header with `===` like the doc for other rules.

Comment (alex-ozdemir) [reply src/expr/proof_rule.h:1056]: Good point. Done in 9dffe397e.

File: src/expr/proof_rule.h [hunk 2]
```diff
@@ -1043,15 +1043,24 @@ enum class PfRule : uint32_t
...
-  // Conclusion: (>< (* k t1) (* k t2))
+  // Conclusion: (>< t1 t2)
   //    where >< is the fusion of the combination of the ><i, (flipping each it
   //    its ki is negative). >< is always one of <, <=
   //    NB: this implies that lower bounds must have negative ki,
   //                      and upper bounds must have positive ki.
-  //    t1 is the sum of the polynomials.
-  //    t2 is the sum of the constants.
+  //    t1 is the sum of the scaled polynomials (sum_i k_i poly_i)
+  //    t2 is the sum of the scaled constants. (sum_i k_i const_i)
```
Comment (ajreynol) [inline src/expr/proof_rule.h:1052]:
```suggestion
  //    t2 is the sum of the scaled constants (k_1 * const_1 + ... + k_n * const_n)
```

File: src/expr/proof_rule.h [hunk 3]
```diff
@@ -1043,15 +1043,24 @@ enum class PfRule : uint32_t
...
   // ---------------------
-  // Conclusion: (>< (* k t1) (* k t2))
+  // Conclusion: (>< t1 t2)
   //    where >< is the fusion of the combination of the ><i, (flipping each it
   //    its ki is negative). >< is always one of <, <=
   //    NB: this implies that lower bounds must have negative ki,
   //                      and upper bounds must have positive ki.
-  //    t1 is the sum of the polynomials.
-  //    t2 is the sum of the constants.
+  //    t1 is the sum of the scaled polynomials (sum_i k_i poly_i)
```
Comment (ajreynol) [inline src/expr/proof_rule.h:1051]:
```suggestion
  //    t1 is the sum of the scaled polynomials (k_1 * poly_1 + ... + k_n * poly_n)
```

**Review by ajreynol — CHANGES_REQUESTED** [review]
Was a bit confused by the documentation.

**Review by ajreynol — APPROVED** [review]

**Pull request merged**